### PR TITLE
feat: add option to ensure kubeconfig can connect to azure

### PR DESCRIPTION
**Pull request by AlisonB319**

### CODE REVIEW

File: client/kubeconfig.go [hunk 1]
```diff
@@ -3,44 +3,82 @@
...
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/client-go/transport"
 	certutil "k8s.io/client-go/util/cert"
 )
 
-func isKubeConfigStillValid(kubeConfigPath string, logger *zap.Logger) (bool, error) {
-	logger.Debug("checking if kubeconfig exists...")
+type KubeClient interface {
+	IsKubeConfigStillValid(kubeConfigPath string) (bool, error)
+	EnsureClusterConnectivity(kubeConfigPath string) error
```
Comment (cameronmeissner) [inline client/kubeconfig.go:23]: I'm actually thinking the better name for this method would be `EnsureAuthentication` or similar

File: client/kubeconfig.go [hunk 2]
```diff
@@ -3,44 +3,82 @@
...
+		return fmt.Errorf("unable to build config: %v", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("unable to create clientset: %v", err)
+	}
+
+	_, err = clientset.Discovery().ServerVersion()
+	if err != nil {
```
Comment (cameronmeissner) [inline client/kubeconfig.go:75]: we still need to double-check how to detect on the particular error we're looking

Comment (AlisonB319) [reply client/kubeconfig.go:75]: Get "
[https://tls-bootstrapping-client-testing-...)
tls: failed to verify certificate: x509: certificate signed by unknown authority

File: client/cmd/main.go [hunk 3]
```diff
@@ -15,11 +15,13 @@ import (
...
-	flagNextProto      = "next-proto"
-	flagAADResource    = "aad-resource"
-	flagVerbose        = "verbose"
+	flagCustomClientID             = "custom-client-id"
+	flagLogFormat                  = "log-format"
+	flagNextProto                  = "next-proto"
+	flagAADResource                = "aad-resource"
+	flagVerbose                    = "verbose"
+	flagKubeconfigPath             = "kubeconfig"
+	ensureKubeClientAuthentication = "ensure-cluster-connectivity"
```
Comment (cameronmeissner) [inline client/cmd/main.go:24]: nit: mismatch

File: client/cmd/main.go [hunk 4]
```diff
@@ -71,6 +73,8 @@ func createBootstrapCommand() *cobra.Command {
 	cmd.Flags().StringVar(&opts.AADResource, flagAADResource, "", "The resource (audience) used to request JWT tokens from AAD for authentication")
 	cmd.Flags().StringVar(&opts.NextProto, flagNextProto, "", "The ALPN Next Protocol value to send within requests to the bootstrap server.")
 	cmd.Flags().StringVar(&opts.LogFormat, flagLogFormat, "json", "Log format: json or text.")
+	cmd.Flags().StringVar(&opts.KubeconfigPath, flagKubeconfigPath, "", "Path to kubeconfig file containing the generated kubelet client certificate.")
 	cmd.Flags().BoolVar(&opts.Verbose, flagVerbose, false, "Enable verbose logging.")
+	cmd.Flags().BoolVar(&opts.EnsureKubeClientAuthentication, ensureKubeClientAuthentication, false, "Ensure kubernetes client authentication before generating the bootstrap token.")
```
Comment (cameronmeissner) [inline client/cmd/main.go:78]: nit: `... before generating a new certificate`

**Review by cameronmeissner — APPROVED** [review]

**Pull request merged**